starters/express-typescript-typeorm-postgres/src/db/datasource.ts

```diff
@@ -12,6 +11,33 @@ export const dataSource = new DataSource({
   password: process.env.DATABASE_PASSWORD,
   database: process.env.DATABASE_NAME,
   entities: ['**/*.entity.js'],
-  logging: IS_DATABASE_LOGGING_ENABLED,
-  synchronize: true,
+  logging: DATABASE_ENABLE_LOGGING,
+  synchronize: DATABASE_ENABLE_SYNC,
 });
+
+const DATABASE_RETRY_COUNT = process.env.DATABASE_RETRY_COUNT
+  ? parseInt(process.env.DATABASE_RETRY_COUNT)
+  : 25;
+
+const DATABASE_RETRY_INTERVAL_MS = process.env.DATABASE_RETRY_INTERVAL_MS
+  ? parseInt(process.env.DATABASE_RETRY_INTERVAL_MS)
+  : 2000;
+
+export async function initialiseDataSource(retries = DATABASE_RETRY_COUNT): Promise<boolean> {
+  return dataSource
+    .initialize()
+    .then(() => true)
+    .catch((err) => {
+      const remainingRetries = retries - 1;
+      console.warn(`Could not connect to the database, retrying ${remainingRetries} more time(s)`);
+      if (remainingRetries === 0) {
+        console.error(`Error during Data Source initialisation:`, err);
+        return false;
+      }
+      return new Promise((resolve) => {
```

Comment:
I think you will need to handle the reject here as well, no?
```ts
return new Promise((resolve, reject) => {
  setTimeout(() => {
    initialiseDataSource(remainingRetries).then(resolve, reject)
  }, DATABASE_RETRY_INTERVAL_MS);
});
```

Reply:
I don't think so, because it is recursion, and the `initialiseDataSource` method calls the `initialiseDataSource` method which always resolves.
In the `.catch()` callback it returns early, when the `remainingRetries === 0` condition is true and AFAIK returning and not throwing in a catch resolves as a promise.

Reply:
Ah yes I see